lib/stat_tracker.rb

```diff
@@ -35,19 +35,12 @@ def total_goals_per_game
     end
   end
 
-  def array_of_total_goals_per_season
-    @games.reduce(Hash.new { |h, k| h[k] = [] }) do |result, game|
-      result[game.season] << game.away_goals + game.home_goals
-      result
-    end
-  end
-
   def total_goals_per_season
-    result = {}
-    array_of_total_goals_per_season.each do |season, goals|
-      result[season] = goals.sum
+    @games.reduce({}) do |result, game|
+      result[game.season] = 0 if !result[game.season]
+      result[game.season] += game.away_goals + game.home_goals
+      result
```

Comment:
Nice Leah! Another way to do this and avoid the guard statement is with `.reduce(Hash.new(0))`. I think is perfect tho! No need to change.

Reply:
Ooh, good call! I think having a default value of 0 here is more readable so I ended up updating it. Thanks for the suggestion!